RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractJumpLists.java

```diff
@@ -0,0 +1,233 @@
+import org.apache.poi.poifs.filesystem.POIFSFileSystem;
+import org.openide.util.NbBundle.Messages;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.casemodule.services.FileManager;
+import org.sleuthkit.autopsy.coreutils.JLNK;
+import org.sleuthkit.autopsy.coreutils.JLnkParser;
+import org.sleuthkit.autopsy.coreutils.JLnkParserException;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.datamodel.ContentUtils;
+import org.sleuthkit.autopsy.ingest.DataSourceIngestModuleProgress;
+import org.sleuthkit.autopsy.ingest.IngestJobContext;
+import org.sleuthkit.autopsy.ingest.IngestServices;
+import org.sleuthkit.autopsy.ingest.ModuleContentEvent;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.DerivedFile;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskData;
+
+/**
+ * Extract the LNK files from the jumplists and save them to ModuleOutput\RecentActivity\Jumplists
+ * and then add them back into the case as a dervived file.
+ */
+final class ExtractJumpLists extends Extract {
+
+    private static final Logger logger = Logger.getLogger(ExtractJumpLists.class.getName());
+
+    private IngestJobContext context;
+
+    private static final String JUMPLIST_TSK_COMMENT = "Jumplist File";
+    private static final String RA_DIR_NAME = "RecentActivity"; //NON-NLS
+    private static final String MODULE_OUTPUT_DIR = "ModuleOutput"; //NON-NLS
+    private static final String AUTOMATIC_DESTINATIONS_FILE_DIRECTORY = "%/AppData/Roaming/Microsoft/Windows/Recent/AutomaticDestinations/";
+    private static final String JUMPLIST_DIR_NAME = "jumplists"; //NON-NLS
+    private static final String VERSION_NUMBER = "1.0.0"; //NON-NLS
+    private String moduleName;
+    private FileManager fileManager;
+    private final IngestServices services = IngestServices.getInstance();
+
+    @Messages({
+        "Jumplist_module_name=Windows Jumplist Extractor",
+        "Jumplist_adding_extracted_files_msg=Chrome Cache: Adding %d extracted files for analysis."
+    })
+    ExtractJumpLists() {
+        super(Bundle.Jumplist_module_name());
+    }
+
+    @Override
+    void process(Content dataSource, IngestJobContext context, DataSourceIngestModuleProgress progressBar) {
+
+        this.context = context;
+        moduleName = Bundle.Jumplist_module_name();
+        fileManager = currentCase.getServices().getFileManager(); 
+        long ingestJobId = context.getJobId();
+
+        List<AbstractFile> jumpListFiles = extractJumplistFiles(dataSource, ingestJobId);
+
+        if (context.dataSourceIngestIsCancelled()) {
+            return;
+        }
+
+        List<AbstractFile> derivedFiles = new ArrayList<>();
+        String derivedPath = null;
+        String baseRaTempPath = RAImageIngestModule.getRATempPath(Case.getCurrentCase(), dataSource.getName() + "-" + JUMPLIST_DIR_NAME, ingestJobId);
```

Comment:
I'd suggest not using the data source name here

Reply:
Changed it to data source id instead.